tests/replication/tcp/streams/test_typing.py

```diff
@@ -75,3 +76,65 @@ def test_typing(self):
         row = rdata_rows[0]
         self.assertEqual(room_id, row.room_id)
         self.assertEqual([], row.user_ids)
+
+    def test_reset(self):
+        """
+        Test what happens when a typing stream resets.
+
+        This is emulated by jumping the stream ahead, then reconnecting (which
+        sends the proper position and RDATA).
+        """
+        typing = self.hs.get_typing_handler()
+
+        room_id = "!bar:blue"
+
+        self.reconnect()
+
+        typing._push_update(member=RoomMember(room_id, USER_ID), typing=True)
+
+        self.reactor.advance(0)
+
+        # We should now see an attempt to connect to the master
+        request = self.handle_http_replication_attempt()
+        self.assert_request_is_get_repl_stream_updates(request, "typing")
+
+        self.test_handler.on_rdata.assert_called_once()
+        stream_name, _, token, rdata_rows = self.test_handler.on_rdata.call_args[0]
+        self.assertEqual(stream_name, "typing")
+        self.assertEqual(1, len(rdata_rows))
+        row = rdata_rows[0]  # type: TypingStream.TypingStreamRow
+        self.assertEqual(room_id, row.room_id)
+        self.assertEqual([USER_ID], row.user_ids)
+
+        # Jump the stream ahead manually, the state of the master is not
+        # modified, however.
+        self.hs.get_tcp_replication().send_command(
+            PositionCommand("typing", "master", 100)
+        )
```

Comment:
We talked about this on in #synapse-dev:matrix.org and I wanted to copy the result here:

> Might be worth just doing some `_push_update` in a loop to advance the stream instead

This would avoid trigger a replication request when the `POSITION` comes through. The data being sent doesn't matter, just that we want to manually push the token forward before resetting it.

Reply:
@erikjohnston I took a look at this and I don't think doing `_push_update` will work -- the point was that I need a way to reset the stream at some point and I've been unable to find that. Sending a `POSITION` command that jumps it ahead implicitly (on the next update) jumps it back, but isn't ideal.

So I think we can do a bunch of `_push_update` commands here, but I'm not sure how to then "reset" the position of the worker?